include/aws/s3/private/s3_request.h

```diff
@@ -144,7 +144,7 @@ struct aws_s3_request_metrics {
         /* The host address of the request. */
         struct aws_string *host_address;
         /* The the request ID header value. */
-        struct aws_string *request_id;
+        struct aws_string *request_attempt_id;
```

Comment:
nope, let's keep this as the request id. Since it's a common name for server to fetch server side logs.